src/payments/services/coupon.service.ts

```
@@ -251,12 +257,31 @@ export class CouponService {
-    // Check country restriction
-    if (coupon.countryId && coupon.countryId !== dto.countryId) {
-      return {
-        isValid: false,
-        error: 'Coupon is not valid for this country',
-      };
+    // Check country restriction: coupon stores countryId; user's country comes from users table.
+    // Resolve user's country to countries table id and compare with coupon.countryId.
+    if (coupon.countryId) {
+      const user = await this.userRepository.findOne({
+        where: { userId: dto.userId },
+        select: ['userId', 'country'],
```

ESLint

[error] Replace `'userId',·'country'` with `"userId",·"country"`

(prettier/prettier)